arelle/plugin/validate/NL/rules/nl_kvk.py

```diff
@@ -505,17 +506,15 @@ def rule_nl_kvk_3_5_2_2(
     NL-KVK.3.5.2.2: All tagged text facts MUST be provided in at least the language of the report.
     """
     reportXmlLang = pluginData.getReportXmlLang(val.modelXbrl)
-    factsWithWrongLang = set()
-    for fact in val.modelXbrl.facts:
-        if (fact is not None and
-                fact.concept is not None and
-                fact.concept.type is not None and
-                fact.concept.type.isOimTextFactType and
-                fact.xmlLang != reportXmlLang):
-            factsWithWrongLang.add(fact)
-    if len(factsWithWrongLang) > 0:
-        yield Validation.error(
-            codes='NL.NL-KVK.3.5.2.2.taggedTextFactOnlyInLanguagesOtherThanLanguageOfAReport',
-            msg=_('Tagged text facts MUST be provided in the language of the report.'),
-            modelObject=factsWithWrongLang
-        )
+    filtered_facts = [f for f in val.modelXbrl.facts if f.concept is not None and
+                      f.concept.type is not None and
+                      f.concept.type.isOimTextFactType and
+                      f.context is not None]
+    factGroups = getHashEquivalentFactGroups(filtered_facts)
+    for fgroup in factGroups:
```

Comment:
This code can't use `getHashEquivalentFactGroups` like this.  Hash collisions need to be handled somehow.

Pre-existing, it seems like this API is easy to misuse, so it should probably be hidden.  It seems like it would be best to have a higher-level key object so that some method like this can return precise groups, or at least a method that returns `list[tuple[tuple[concept,context,unit],list[fact]]]`.  `getAspectEqualFacts` is almost that (somewhat confusingly named), but it would need an option to stop considering lang.